### PR TITLE
feat: Providing list length info for PopFront, PopBack, Remove, Retain, and Erase Responses.

### PR DESCRIPTION
**Pull request by honeyAcorn**
[Problem]
Clients may get list length after list modification command; resulting in race-condition. 
For more information, please refer to:
https://github.com/momentohq/cacheservice/issues/1197

[Solution]
Including list length in PopFront, PopBack, Remove, Retain, Erase responses.

### CODE REVIEW

**Review by wandaitzuchen — APPROVED** [review]
LGTM! I'll wait for @cprice404 to take a look before merging.

File: proto/cacheclient.proto
Comment (mjindalmotiv8) [inline proto/cacheclient.proto:405]: @cprice404 - can you confirm that these responses will not break SDK's?

Comment (cprice404) [reply proto/cacheclient.proto:405]: I'm 90% sure that this is fine.  They will just not be deserialized by the SDKs until we update the SDK protos.

But, to be safe, we should just test by running integration tests in a dev cell and/or alpha when we have a working version of the corresponding ms2/mr2 changes.

**Review by cprice404 — APPROVED** [review]
These changes look fine to me but y'all should figure out an order of operations for how you want to test them with ms2+mr2+existing SDK integration tests and decide when to merge based on that.

**Comment by wandaitzuchen** [comment]
@HoneyAcron please hold off merging this until later. We should test SDKs out when you have the backend implementations ready in your dev cell.

**Comment by HoneyAcron** [comment]
>

**Comment by HoneyAcron** [comment]
I will leave this open/hold for now until we have fully tested it with the backend implementation. I accidentally close this PR so now I'm reopening it.

**Comment by mjindalmotiv8** [comment]
In accordance with Chris's comments - here is my feedback on how to go about this - leave this proto open

1. Work on MS2 and merge the MS2 change
2. Work on MR2 with local version of client proto
3. Test that SDK is not broken in dev cell
4. Commit this proto change if SDK works
5. Commit the MR2 change
6. Work on the SDK changes

**Comment by HoneyAcron** [comment]
Added Kenny's feedback. For responses in Erase, Remove, Retain, we need to return Found or Missing and include the length in Found. This is because previously user don't need to know if the operation is successful. If successful, it will just delete/modify. However, if the key/value doesn't exist, it simply won't do anything, so no need to provide a response.

Now that we are responding with list_length, if user get a 0 for list length, they can't really tell if it's the real length or the modification failed. Thus adding new fields.

File: proto/cacheclient.proto
Comment (cprice404) [inline proto/cacheclient.proto:456]: @kvcache same story here w/rt backward compat, right?  Existing SDKs will just ignore this data and continue to operate as they did before, yes?

File: proto/cacheclient.proto
Comment (cprice404) [inline proto/cacheclient.proto:481]: this stuff is going to be a little more interesting w/rt the SDKs.  Today, we have several 1.0 SDKs that include a ListRemoveValue function, and they currently have only `Success` and `Failure` response types.

It's easy enough to add the new list length value to the `Success` type, but changing the types be a breaking change.  We could theoretically do something like add a `Miss` response type and leave `Success` in place, mapped to `_Found`, but that would be a little inconsistent with our existing APIs.

@malandis @kvcache I'm interested in thoughts on this.

Comment (kvcache) [reply proto/cacheclient.proto:481]: I think it's not a huge issue to expose found/missing through success to the user. In fact, I recommend mapping it to `Success { list_length: 0 }` on the SDK Success object to start out with, and when requested you can add a method `fn was_found() -> bool` (or whatever name you like) to Success without breaking anyone.

Comment (cprice404) [reply proto/cacheclient.proto:481]: or I guess `listLength` could be of type `int?` ?

**Comment by cprice404** [comment]
If @malandis is on board with condensing Found/Missing => Success in the sdks then I'm +1 on this

File: proto/cacheclient.proto
Comment (kvcache) [reply proto/cacheclient.proto:481]: You're talking about `momento_sdk::Response::Success::listLength`? You could do that too. Imo it's more annoying than letting it be `int` and having to ask for "did it do the delete?" separately. I'd hazard that the 95% use case is only concerned with `int`. As far as the protos go, I'm just trying to help keep a non-fire-drill avenue open for helping the sdk's serve the remaining vocal 5%, should they speak up.

Comment (cprice404) [reply proto/cacheclient.proto:481]: yeah you're right re: 90% case.

**Comment by honeyAcorn** [comment]
@cprice404 I've tested my [MR2](https://github.com/momentohq/cacheservice/pull/1318) and [MS2](https://github.com/momentohq/storage-store/pull/176) code changes with the client-sdk-dotnet integration test and it's backward compatible. Is there any other sdk that you think we can test it with before we merge this PR?

**Review by cprice404 — APPROVED** [review]

**Pull request merged**
